### PR TITLE
Better manage transform service lifecycle for tests

### PR DESCRIPTION
**Pull request by chamikaramj**
Wait for the transform service to start/stop in tests.

Start unique transform service instance per test task.

Cleanup transform service temp directory after during the clanup phase of the test task.

Make sure to build the external SDK harnesses referred to by the transform service.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by chamikaramj** [comment]
Run TransformService_Direct PostCommit

**Comment by chamikaramj** [comment]
Run TransformService_Direct PostCommit

**Comment by chamikaramj** [comment]
Run TransformService_Direct PostCommit

**Comment by chamikaramj** [comment]
Run TransformService_Direct PostCommit

**Comment by chamikaramj** [comment]
Run TransformService_Direct PostCommit

**Comment by chamikaramj** [comment]
R: @Abacn or @damccorm :)

This should fix the perma red new test suite https://ci-beam.apache.org/job/beam_PostCommit_TransformService_Direct/

**Comment by chamikaramj** [comment]
Run Java_Pulsar_IO_Direct PreCommit

**Comment by chamikaramj** [comment]
Run Portable_Python PreCommit

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -2807,9 +2810,17 @@ class BeamModulePlugin implements Plugin<Project> {
         throw new GradleException(exceptionMessage)
       }
 
+      // Transform service delivers transforms that refer to SDK harness containers with following sufixes.
+      def transformServiceJavaContainerSuffix = 'java11'
```
Comment (Abacn) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:2814]: what caused this test requiring two java/python containers to work? Could it be able to clean up to make the test working for single java / py sdk container?

**Review by Abacn — COMMENTED** [review]
Thanks for the fix, commented below

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 1 above)
Comment (chamikaramj) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:2814]: So this setup method is (or rather will be) shared by PythonFromJava and JavaFromPython tests.

For external transforms, Transform Service will require Python3.8 and Java11 containers (transformServicePythonContainerSuffix and transformServiceJavaContainerSuffix).

The pipeline SDK, we could run tests with multiple Java and Python versions hence we have to build such containers based on the test ( pythonContainerSuffix and javaContainerSuffix).

For example, currently BT test is run using Python3.8 and 3.11 for pipeline SDK and Java 11 for external SDK.

Comment (Abacn) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:2814]: Thanks for the explanation. I see, for a single slang test, there are at least two SDK container is working

Comment (Abacn) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:2814]: (though it is generally nice to have these version not hard coded)

**Review by Abacn — APPROVED** [review]

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 1 above)
Comment (chamikaramj) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:2814]: Thanks. Beam ships with these specific versions (Python 3.8 and Java 11) of expansion service containers. I hope to add documentation on developing expansion service containers with additional language versions for users who need that.

**Pull request merged**